engineapi/instance_manager.go

```diff
@@ -56,12 +65,53 @@ func NewInstanceManagerClient(im *longhorn.InstanceManager) (*InstanceManagerCli
 	if im.Status.CurrentState != longhorn.InstanceManagerStateRunning || im.Status.IP == "" {
 		return nil, fmt.Errorf("invalid Instance Manager %v, state: %v, IP: %v", im.Name, im.Status.CurrentState, im.Status.IP)
 	}
+	// HACK: TODO: fix me
+	endpoint := "tcp://" + imutil.GetURL(im.Status.IP, InstanceManagerDefaultPort)
+
+	initTLSClient := func() (*imclient.ProcessManagerClient, error) {
+		// check for tls cert file presence
+		tlsDir := types.TLSDirectoryInContainer
+		pmClient, err := imclient.NewProcessManagerClientWithTLS(endpoint,
+			filepath.Join(tlsDir, "ca.crt"),
+			filepath.Join(tlsDir, "tls.crt"),
+			filepath.Join(tlsDir, "tls.key"),
+			"longhorn-backend.longhorn-system",
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load Instance Manager Client TLS files Error: %w", err)
+		}
+
+		if _, err = pmClient.VersionGet(); err != nil {
+			return nil, fmt.Errorf("failed to check check version of Instance Manager Client with TLS connection Error: %w", err)
+		}
+
+		return pmClient, nil
+	}
+
+	pmClient, err := initTLSClient()
+	if err != nil {
+		// fallback to non tls client, there is no way to differentiate between im versions unless we get the version via the im client
```

Comment:
I would suggest having a log for fallback behavior, it's better to understand the context at runtime.

Reply:
Generally agree adding the logrus ref here kinda sucks though.
I didn't want to change this to much before the proxy PR.
Afterwards I intended to add a client factory so we can have im client mock for unit tests.